wails-frontend/src/pages/backupKeyPairs.tsx

```diff
@@ -53,72 +49,80 @@ function CopyKey() {
 }
 
 function BackupKeyPairs() {
-  const navigate = useNavigate();
   const form = useRef(null);
   const { state } = useLocation();
+  const navigate = useNavigate();
 
   const req: promptRequest = state.req;
-  const { backupReq } = promptAction;
-
-  const isExportAction = req.Action === backupReq;
 
-  const [show, setShow] = useState<boolean>(false);
+  const [privateKey, setPrivateKey] = useState<string | undefined>(undefined);
   const [error, setError] = useState<IErrorObject | null>(null);
 
   function validate(e: SyntheticEvent) {
     const formObject = parseForm(e);
     const { password } = formObject;
 
-    if (!hasMoreThanFiveChars(password)) {
-      setError({ password: 'Password must have at least 5 characters' });
-      return false;
-    } else if (!password || !password.length) {
+    if (!password || !password.length) {
       setError({ password: 'Password is required' });
       return false;
     }
     return true;
   }
 
-  function save(e: SyntheticEvent) {
-    const formObject = parseForm(e);
-    const { password } = formObject;
-
-    EventsOnce(
-      events.promptResult,
-      handleApplyResult(navigate, req, setError, isExportAction),
-    );
-
-    return ApplyPassword(password);
+  async function handleResult(result: promptResult) {
+    if (!result.Success) {
+      setError({ error: getErrorMessage(result.CodeMessage) });
+      navigate('/failure', {
+        state: { req },
+      });
+    }
+    setPrivateKey(result.Data);
   }
 
   async function handleSubmit(e: SyntheticEvent) {
     e.preventDefault();
     if (!validate(e)) return;
 
-    save(e);
-    // setShow(true); when backend is ready
-    setShow(false);
+    const formObject = parseForm(e);
+    const { password } = formObject;
+
+    EventsOnce(events.promptResult, handleResult);
```

Comment:
ref: https://wails.io/docs/reference/runtime/events/#eventsonce

EventsOnce doesn't return a Promise, so it's sync function, but we need handleSubmit to be async because we await SendPromptInput